src/errorLogging/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { logException } from '@wordpress/react-native-bridge';
+
+// Setting Error handler to send exception. This implemenetation is based on Sentry React Native SDK:
+// https://github.com/getsentry/sentry-react-native/blob/adfb66f16438dfd98f280307844778c7291b584b/src/js/integrations/reactnativeerrorhandlers.ts#L187-L262
+export default () => {
+	const errorUtils = global.ErrorUtils;
+	const defaultHandler =
+		errorUtils.getGlobalHandler && errorUtils.getGlobalHandler();
+
+	let handlingFatal = false;
+	errorUtils.setGlobalHandler( ( error, isFatal ) => {
+		// We want to handle fatals, but only in production mode.
+		const shouldHandleFatal = isFatal && ! __DEV__;
+		if ( shouldHandleFatal ) {
+			if ( handlingFatal ) {
+				// eslint-disable-next-line no-console
+				console.warn(
+					'Encountered multiple fatals in a row. The latest:',
+					error
+				);
+				return;
+			}
+			handlingFatal = true;
+		}
+
+		// TODO: Set severity and exception mechanism.
+		// https://github.com/getsentry/sentry-react-native/blob/adfb66f16438dfd98f280307844778c7291b584b/src/js/integrations/reactnativeerrorhandlers.ts#L235-L239
```

Comment:
The severity and exception mechanism should be added in the [`logException` function](https://github.com/WordPress/gutenberg/blob/12aff798a3c3f0fb598f520a7901b86ed337cb9c/packages/react-native-bridge/index.js#L537-L562) when passing a parameter (e.g. `unhandled`).

Reply:
The mechanism is inserted in the Crash logging service ([reference](https://github.com/Automattic/Automattic-Tracks-iOS/blob/decdafc1c8b08ff83935cccc8880b0809ca9e5c6/Sources/Remote%20Logging/Crash%20Logging/SentryEventJSException.swift#L65-L68)).